src/bust.py

```diff
@@ -288,6 +271,9 @@ def ffmpeg_post_hook(e: BaseException = None):
         bot_member = self.current_channel.guild.get_member(self.client.user.id)
         await bot_member.edit(nick=new_nick)
 
+        # Wait for song to finish playing
+        await play_lock.acquire()
```

Comment:
What is the point of this lock?

Reply:
It waits for the song to finish playing before returning from the function, so that code like
```python
play_song(0)
play_song(1)
```
plays two songs in a row instead of trying to start playing two songs at almost the same time. It's necessary because `voice_client.play` is not async, it actually starts playback in a new thread.